internal/saltfilter.go

```diff
@@ -0,0 +1,86 @@
+package internal
+
+import (
+	"fmt"
+	"strconv"
+)
+
+// Those suggest value are all set according to
+// https://github.com/shadowsocks/shadowsocks-org/issues/44#issuecomment-281021054
+// Due to this package contains various internal implementation so const named with DefaultBR prefix
+const (
+	DefaultBRCapacity          = 1e6
+	DefaultBRFalsePositiveRate = 1e-6
+	DefaultBRSlot              = 10
+)
+
+// Support override ability while in compiling via `-ldflags -X` and
+// use a private variable to prevent read/write outside this package
+var (
+	brCapacity          string
+	brFalsePositiveRate string
+	brSlot              string
+)
+
+// A shared instance used for checking salt repeat
+var saltfilter *BloomRing
+
+func init() {
+	var (
+		finalCapacity = DefaultBRCapacity
```

Comment:
Is there anyway to change the value from the command line flags?

Reply:
Absolutely, but I thought about this: low-level flags maybe more suitable to expose to people who known this what to do and how to modify(to make it works as expect), and keep user-end flags as Shadowsocks well known provides.

Reply:
Makes sense. Maybe we could change it via environment variables?

Reply:
A good idea as well, with `SHADOWSOCKS_` prefix or not?

Reply:
- `SHADOWSOCKS_SF_CAPACITY`
- `SHADOWSOCKS_SF_FPR`
- `SHADOWSOCKS_SF_SLOT`

Reply:
Looks good to me. I'm debating if we should default the filter on or off. Maybe off by default?

Reply:
In my view, it's better to be enabled by default as it can 'protects' server by withstand reuse detection.

Reply:
Makes sense.